### ISSUE TITLE
Grafana angular deprecation in grafana 11

### ISSUE THREAD

**Issue by NissesSenap** [kubernetes-monitoring/kubernetes-mixin#898]
Hey all

In Grafana 11, angular dashboards will be deprecated
https://grafana.com/docs/grafana/latest/developers/angular_deprecation

Grafana 10.4 just got released, and I bumped my grafana instances and all of K8s mixins shows warnings that it's time to move away from the old Graph solution.

![image](https://github.com/kubernetes-monitoring/kubernetes-mixin/assets/14774899/31431852-59e6-4b1e-954a-acd3c3baef7b)

![image](https://github.com/kubernetes-monitoring/kubernetes-mixin/assets/14774899/cbc48ba9-6631-4e46-8bae-29787a7eae6d)

In short, we need to update our panels, or it will become a blocker to upgrade to grafana 11 when that gets released during GrafanCon in  April.

**Comment by NissesSenap** [comment]
Sadly, I never learned jsonnet, so I don't feel I can help out with the issue.

**Comment by jkroepke** [comment]
> In short, we need to update our panels, or it will become a blocker to upgrade to grafana 11 when that gets released during GrafanCon in April.

It's not an blocker. You can already disable AngularJS in grafana via `security.angular_support_enabled = false` which triggers the auto migration of the table panel.

**Comment by NissesSenap** [comment]
I did, not know that @jkroepke , that is really good to know. 
But the most beautiful solution, but if it works it's great. Might be worth documenting as a workaround.

**Comment by jkroepke** [comment]
> But the most beautiful solution, but if it works it's great.

Important to mention is that deep links wont work anymore. At kube-promethues-stack, we implement a toggle that fixes deeplinks on auto-migrated table panels.

Ref: https://github.com/kubernetes-monitoring/kubernetes-mixin/pull/884

**Comment by marians** [comment]
What is the proper way to migrate the panels here in the repo?

In https://github.com/kubernetes-monitoring/kubernetes-mixin/blob/release-0.12/dashboards/resources/namespace.libsonnet#L163 for example I see the `queryPanel` function from github.com/grafana/jsonnet-libs/grafana-builder/grafana.libsonnet called. Does that function have to be updated then?

Or do we have to migrate from `queryPanel` to [`timeseriesPanel`](https://github.com/grafana/jsonnet-libs/blob/master/grafana-builder/grafana.libsonnet#L233)?

And if [`tablePanel`](https://github.com/grafana/jsonnet-libs/blob/master/grafana-builder/grafana.libsonnet#L312) produces a deprecated result, what is the replacement for that?

**Comment by marians** [comment]
I just found https://github.com/kubernetes-monitoring/kubernetes-mixin/pull/890 by @Duologic which already migrated time series and stat panels in https://github.com/Duologic/kubernetes-mixin/blob/master/dashboards/apiserver.libsonnet

And I just noticed that https://github.com/grafana/grafonnet-lib is archived.

**Comment by DrFaust92** [comment]
@jkroepke are you sure it will be auto migrated with https://grafana.com/docs/grafana/latest/setup-grafana/configure-grafana/#angular_support_enabled? it seems it will just make it not load according to docs

**Comment by jkroepke** [comment]
Yeah, people confirm it: https://github.com/prometheus-community/helm-charts/issues/4398#issuecomment-2024551358

I'm also running Grafana instances with `[angular_support_enabled](https://grafana.com/docs/grafana/latest/setup-grafana/configure-grafana/#angular_support_enabled)`. The issue is that deep-links are not migrated well and we have provide a fix for that: https://github.com/prometheus-community/helm-charts/pull/4297

**Comment by skl** [comment]
👋 here's the 2nd dashboard to be ported, as far as I'm aware, reviews and comments welcome!

- #904

**Comment by skl** [comment]
👋 number 3:

- #907

**Comment by EvertonSA** [comment]
thanks a lot to all working on this, if there is something i can help, please tag me

**Comment by skl** [comment]
I think this issues can be closed with the final dashboards ported here:

- #931

Many thanks to @Duologic and @rgeyer for contributing and for @povilasv for handling all the reviews and merging! Thanks also to the people commenting on this thread, it was a great point of reference to see everything in one place ❤️

**Comment by skl** [comment]
@ehddnko make sure that you have set the correct scrape interval to match your data in Grafana > Connections > Data sources > (your Prometheus data source) > Settings > Advanced Settings > Internal Behavior > Scrape interval (e.g. I have set mine to `60s`).

Grafana needs this information to be set manually in order for `$__rate_interval` to work correctly, as it cannot be discovered automatically by the data source.

<img width="590" alt="Screenshot 2024-05-13 at 13 11 45" src="https://github.com/kubernetes-monitoring/kubernetes-mixin/assets/514343/8f3444f7-8bbd-4e2e-94fa-0075141fc355">

More info here:

- https://grafana.com/blog/2020/09/28/new-in-grafana-7.2-__rate_interval-for-prometheus-rate-queries-that-just-work/

**Comment by ehddnko** [comment]
@skl Thank you for your advice! I found that the scrape interval was set to `15s` in the Mimir data source. After changing this scrape interval, it works correctly.

**Comment by skl** [comment]
@povilasv can this issue be closed to indicate it's resolved? As it's now being referenced by dependent projects experiencing the same.

**Comment by jkroepke** [comment]
Close it!

Upcoming bugs can be a individual issue.
